cdap-tms/src/main/java/io/cdap/cdap/messaging/service/CoreMessagingService.java

```diff
@@ -438,22 +438,27 @@ private Map<String, String> createDefaultProperties() {
     return properties;
   }
 
+  private TopicMetadata getTopic(TopicId topicId) throws TopicNotFoundException, IOException {
+    try {
+      return topicCache.get(topicId);
+    } catch (ExecutionException e) {
```

Comment:
Make `getTopicMetadataProperties` call this method, so that we don't have code duplication.